openstef/model_selection/model_selection.py

```diff
@@ -173,9 +174,8 @@ def split_data_train_validation_test(
         test_data = data_[start_date_test:]
         train_val_data = data_[:start_date_test]
     else:
-        start_date_val = start_date + np.round(number_indices * test_fraction) * delta
-        test_data = data_[:start_date_val]
-        train_val_data = data_[start_date_val:]
+        test_data = data_.copy(deep=True)
```

Comment:
I remember disusing this but on a closer look I do not understand this anymore. For the split_test_train_validation data I would expect three different datasets to come out (may also be empty datasets).

What is the problem again with just using `pd.concat([train_data, validation_data])` on the place were you want to use the test_data_set?

Reply:
The main reason is related to the use of custom split function. If you implement your own split procedure it seems very akward to design it with this concat operation in mind. Moreover it seems very sensible to compute scores/metrics on the test set.   
My changes do not affect the fact that the split function returns 3 datasets (they are just overlapping when `back_test=False`).   
The second issue is the fact that this concat can introduce weird behaviours when the order of train and val are shuffled. This has no consequences for pointwise models like xbgm/lightgbm because lag features has been computed before splitting. But with ARIMA (or other ar models like prophete) it can be very painfull, that why we prefer to keep this operation behind the split function.